### ISSUE TITLE
`clickhouse-client` does not check domain name of the server's SSL certificate

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#35504]
It connects successfully even if domain name mismatch.

The issue reported by @mlex

**Comment by mlex** [comment]
The domain name validation can be enabled by adding this to the client config (e.g. `/etc/clickhouse-client/config.xml`):
```
  <openSSL>
    <client>
            <extendedVerification>true</extendedVerification>
    </client>
  </openSSL>
```

Then clickhouse-client verifies the domain name:

```
$ ./clickhouse client --host xxxxxxx.eu-west-1.compute.amazonaws.com --secure --port 9440
ClickHouse client version 22.4.1.1301 (official build).
Connecting to xxxxxxx.eu-west-1.compute.amazonaws.com as user default.
Code: 210. DB::NetException: Certificate validation error: Unacceptable certificate from xxxxxxx.eu-west-1.compute.amazonaws.com: application verification failure (xxxxxxx.eu-west-1.compute.amazonaws.com:9440). (NETWORK_ERROR)

[ec2-user@ip-172-31-28-243 ~]$ ./clickhouse client --host correct-hostname.mydomain.com --secure --port 9440
ClickHouse client version 22.4.1.1301 (official build).
Connecting to correct-hostname.com:9440 as user default.
Connected to ClickHouse server version 22.4.1 revision 54455.
```

**Comment by alexey-milovidov** [comment]
Thank you!
We can enable it by default, but it will be a change in behavior that users may not expect.
Should we wait for https://github.com/ClickHouse/ClickHouse/issues/35972?

**Comment by mlex** [comment]
Yes, I think we should wait. I wouldn't be surprised if users depend on this default behaviour.

**Comment by mlex** [comment]
I added the current default (and a small info that  `extendedVerification` means "verify hostname") to the documentation: https://github.com/ClickHouse/ClickHouse/pull/36191

**Comment by alexey-milovidov** [comment]
- clickhouse-client should validate the domain name by default; otherwise, an attacker controlling the network, such as a malicious ISP or malicious datacenter, can trick a user into exposing their queries;
- it is not possible to apply the compatibility setting from the server side here; however, it can be provided in the client configuration, but it does not make sense as it's possible to explicitly configure the level of validation;
- we should not worry too much about this behavior change, as HTTPS clients already validate everything, plus a workaround is simple if needed.

**Comment by asad-awadia** [comment]
@alexey-milovidov is localhost an exception? I am using a cert with a CN and SAN and then i connect via clickhouse client cli on --host localhost and it works fine -- should it not error out?

**Comment by kaijaytu** [comment]
Hi, I'd like to work on this if it's still available.

Based on the discussion, the plan is to enable extendedVerification by default in clickhouse-client so that SSL hostname verification is on out of the box. I'll look into the SSL context setup in the client code and make this the default, ensuring users can still disable it via config if needed.

Could you assign this to me?
